tests/template_backends/test_jinja2.py

```diff
@@ -21,3 +21,21 @@ class Jinja2Tests(TemplateStringsTests):
     engine_class = Jinja2
     backend_name = 'jinja2'
     options = {'keep_trailing_newline': True}
+
+    def test_self_context(self):
+        """
+        #24538 -- Using 'self' in the context should not throw errors
+        """
+        engine = Jinja2({
+            'DIRS': [],
+            'APP_DIRS': False,
+            'NAME': 'django',
+            'OPTIONS': {},
+        })
```

Comment:
Can you reuse the existing `self.engine` instead of defining a new one? That would keep the test short and focused. (If there's a good reason to define a new engine, sorry, I missed it.)